os-fpga/FOEDAG
# GTKWave: Fix interface issues and change non-error msg to status msg.

> ### Motivate of the pull request
> Fix a typo in the wave_refresh command and convert a normal message that gtkwave sends as an error to a status message instead.

> ### Describe the technical details
> #### What is currently done? (Provide issue link if applicable)
> - `wave_refresh` has a typo and calls `gtkwave::reloadFile` when it should be `gtkwave::reLoadFile`
> - `wave_open` sent relative paths to gtkwave which launches from a different bin dir which could result in files not opening.
> - GTKWave reports a file's start and end time at load, but it does it on stderr for some reason so every load prints an error message:
>   ![image](https://user-images.githubusercontent.com/103447061/205103066-0538db5e-fce1-4872-a99a-3bfadcbd29b8.png)

>
> #### What does this pull request change?
> - `wave_refresh` now calls `gtkwave::reLoadFile`
> - `wave_open` now converts relative paths to full paths before passing to GTKWave. If the file doesn't exist, foedag will print an error.
> - Load messages are now treated as status instead of error:
>   ![image](https://user-images.githubusercontent.com/103447061/205103410-7820bc0e-8bf2-41aa-9440-62489d5e658c.png)


> ### Which part of the code base require a change
> 
> - [x] Library: Compiler
> - [ ] Plug-in: <Specify the plugin name>
> - [ ] Engine
> - [ ] Documentation
> - [ ] Regression tests
> - [ ] Continous Integration (CI) scripts

> ### Testing
> - Manually tested in foedag
> - tests/gtkwave_open_bad_path.tcl has been added to verify `wave_open` path fix
> - wave_refresh change will be tested when RG-199 is finished
